api/agent/call.go

```diff
@@ -34,9 +34,8 @@ type Call interface {
 
 	// End will be called immediately after attempting a call execution,
 	// regardless of whether the execution failed or not. An error will be passed
-	// to End, which if nil indicates a successful execution. Any error returned
-	// from End will be returned as the error from Submit.
-	End(ctx context.Context, err error) error
+	// to End, which if nil indicates a successful execution.
+	End(ctx context.Context, err error)
```

Comment:
we need a way to communicate errors back from things here?

Reply:
I don't think so, because two sources of error in call.End(), which are c.handler.Finish() or c.ct.fireAfterCall(). In the old code, we log c.handler.Finish() while return c.ct.fireAfterCall. But in the code base these are all ignored anyway. I felt like it's clearer to just acknowledge it and fixup the interface, but I can also take this change out of this PR if this looks problematic. (Also going forward I think we should also take error returns from fireAfterCalls() since this can cause leaks /issues since due to errors some of fireAfterCalls might not fire especially when multiple of these are registered.)

Reply:
yea, bug definitely snuck in here then. it should get the error from fire after call and hand it to end (for example, if fire after call errors, async may not want to delete the message in finish to retry it) - we don't have great testing around async and it's effectively dead now, yea. i agree about fire after call contract we should probably call all of them and hand off preceding errors, in general it's a bit of a mess with start/beforecall/finish/aftercall